frontend/src/components/widgets/CameraImageInput/CameraImageInput.tsx

```diff
@@ -0,0 +1,499 @@
+      return
+    }
+
+    const prevWidgetValue = widgetMgr.getFileUploaderStateValue(element)
+    if (!_.isEqual(newWidgetValue, prevWidgetValue)) {
+      widgetMgr.setFileUploaderStateValue(element, newWidgetValue, {
+        fromUi: true,
+      })
+    }
+  }
+
+  /**
+   * When the server receives the widget value, it deletes "orphaned" uploaded
+   * files. An orphaned file is any file, associated with this uploader,
+   * whose file ID is not in the file ID list, and whose
+   * ID is <= `newestServerFileId`. This logic ensures that a FileUploader
+   * within a form doesn't have any of its "unsubmitted" uploads prematurely
+   * deleted when the script is re-run.
+   */
+  private createWidgetValue(): FileUploaderStateProto | undefined {
+    if (this.state.newestServerFileId === 0) {
+      // If newestServerFileId is 0, we've had no transaction with the server,
+      // and therefore no widget value.
+      return undefined
+    }
+
+    const uploadedFileInfo: UploadedFileInfoProto[] = this.state.files
+      .filter(f => f.status.type === "uploaded")
+      .map(f => {
+        const { name, size, status } = f
+        return new UploadedFileInfoProto({
+          id: (status as UploadedStatus).serverFileId,
+          name,
+          size,
+        })
+      })
+
+    return new FileUploaderStateProto({
+      maxFileId: this.state.newestServerFileId,
+      uploadedFileInfo,
+    })
+  }
+
+  /**
+   * If we're part of a clear_on_submit form, this will be called when our
+   * form is submitted. Restore our default value and update the WidgetManager.
+   */
+  private onFormCleared = (): void => {
+    this.setState({ files: [] }, () => {
+      const newWidgetValue = this.createWidgetValue()
+      if (newWidgetValue == null) {
+        return
+      }
+
+      this.setState({
+        imgSrc: null,
+      })
+
+      this.props.widgetMgr.setFileUploaderStateValue(
+        this.props.element,
+        newWidgetValue,
+        { fromUi: true }
+      )
+    })
+  }
+
+  public render = (): React.ReactNode => {
+    const { element, widgetMgr } = this.props
+
+    // Manage our form-clear event handler.
+    this.formClearHelper.manageFormClearListener(
+      widgetMgr,
+      element.formId,
+      this.onFormCleared
+    )
+
+    if (!this.state.imgSrc) {
+      return (
+        <div>
+          <Webcam
+            audio={false}
+            ref={this.webcamRef}
+            screenshotFormat="image/jpeg"
+            screenshotQuality={1}
+            onUserMediaError={this.onMediaError}
+            onUserMedia={this.onUserMedia}
+          />
+          <UIButton kind={Kind.PRIMARY} onClick={this.capture}>
+            Capture photo
+          </UIButton>
+
+          {this.state.webcamRequestState === "error" && (
+            <div>Please allow access to Webcam</div>
+          )}
+          {this.state.webcamRequestState === "pending" && (
+            <div>Please allow access to Webcam, Please</div>
```

Comment:
I think it can just be 
"Please allow access to Webcam" 

I think though the product spec wanted to somehow give directions to enable webcam.

Reply:
O, good catch!
I just forgot to delete this :) :)